src/component/handlers/edit/editOnPaste.js

```diff
@@ -146,9 +146,21 @@ function editOnPaste(e: SyntheticClipboardEvent): void {
         this.props.blockRenderMap
       );
       if (htmlFragment) {
-        var htmlMap = BlockMapBuilder.createFromArray(htmlFragment);
-        this.update(insertFragment(this.props.editorState, htmlMap));
-        return;
+        const { contentBlocks, contentState } = htmlFragment;
```

Comment:
Just to be clear, the idea here is that the returned `htmlFragment` has a `ContentState` with the new entities already added to it but not necessarily the relevant ranges of text -- those are included in the `contentBlocks` array?

If so, I'm wondering if `htmlFragment` could instead be something like

```javascript
{
  contentBlocks: Array<ContentBlock>,
  entityMap: EntityMap,
}
```

With `insert-fragment` accepting both a block map and an entity map for insertion.

What do you think?